content/blog/cruxbox/2020-05-06_gsoc_2020_project_adding_xfs_filesystem_support.md

```diff
@@ -0,0 +1,50 @@
++++
+type = "blog"
+title = "GSoC 2020 Project: Adding XFS file system in Haiku"
+author = "CruxBox"
+date = "2020-05-11 23:35:56+05:30"
+tags = ["haiku", "software", "gsoc", "gsoc2020", "file system"]
++++
+
+
+Sorry for the late post, but here we go!
+
+# Introduction:
+
+I am Shubham Bhagat, a sophomore, currently majoring in Computer Science Engineering from Indian Institute of Information Technology, Sricity, India.
+I came across Haiku, last November (2019), while I was looking for an Operating System I could contribute to. My interest to contribute came from a mini college course. I knew Haiku was also part of GSoC for many years now, so getting the chance to work as an intern and learn here was a no brainer. So THANK YOU for selecting me as a mentee!
+
+---
+
+# About the project:
+
+My project proposes to begin support for the v4 of XFS File System. We currently have no support for it. The goal is to get the read-only part setup for now. If time permits we can do more :)
+
+# Project plan:
+
+I was spending time reading a lot of the specs and the code I had written for the past few days(It's been over a month since I did any work because of university commitments). At the moment, I think the initial step is to be able to read small directories. The way the inode manages data is in a hierarchical way depending on the size of the data. For example, directories, depending on the number of entries, could be:
```

Comment:
```suggestion
I was spending time reading a lot of the specs and the code I had written for the past few days (it's been over a month since I did any work because of university commitments). At the moment, I think the initial step is to be able to read small directories. The way the inode manages data is in a hierarchical way depending on the size of the data. For example, directories, depending on the number of entries, could be:
```